src/run.rs

```diff
@@ -241,6 +241,15 @@ impl Heap {
     };
   }
 
+  pub fn from_data(data: Vec<(Ptr, Ptr)>) -> Heap {
```

Comment:
maybe a better way is to impl `Default` in `Heap` so that it returns the `{data: [], next: 1, used: 0, full: false}`. Then change `Heap::new` into `Heap { data: vec(size), ..Heap::default() }`, and remove the `from_data` function, because you'd be able to call `Heap { data, ..Heap::default() }` then

Reply:
If changing the .data manually is ok (Heap.data would need to be pub(crate) then), then i could just use:

```rs
let mut heap = run::Heap::new(0);
heap.data  = self.heap.into_iter().map(|node| node_to_pair(*node)).collect();
```

would this be better?